chia/util/files.py

```diff
@@ -71,8 +73,8 @@ async def write_file_async(file_path: Path, data: Union[str, bytes], *, file_mod
     mode: Literal["w+", "w+b"] = "w+" if type(data) == str else "w+b"
     temp_file_path: Path
     async with tempfile.NamedTemporaryFile(dir=file_path.parent, mode=mode, delete=False) as f:
-        temp_file_path = f.name  # type: ignore[assignment]
-        await f.write(data)  # type: ignore[arg-type]
+        temp_file_path = f.name
+        await f.write(data)
```

Comment:
I twiddled with these just the other day and decided not to bother with them.  It's not obvious how to tie the mode chosen by the type of the incoming data to the type of file object we'll get out.  Plus other bits.  If you want to chat about it we can.  Otherwise feel free to just put the ignores back in.

https://github.com/Chia-Network/chia-blockchain/pull/14506/files

```suggestion
        temp_file_path = f.name  # type: ignore[assignment]
        await f.write(data)  # type: ignore[arg-type]
```

Reply:
Thanks - I tried exactly this locally, but there seems to be a bug in `mypy`. When I replace these ignores, but remove the file from the mypy.ini ignores list, it gives "unused ignore statement" for those two lines.

I think I'll report this to mypy, and close this PR for now.

Reply:
Are you sure you were running mypy v1.0 locally?  That's what introduced the complaints here.